.github/workflows/cml.yaml

```
@@ -1,23 +1,29 @@
 name: CML Report
-on: push
+on: [push, pull_request]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.head.sha || github.sha }}
+  cancel-in-progress: true
 jobs:
   run:
-    runs-on: [ubuntu-latest]
+    continue-on-error: ${{ github.ref_name != 'main' }}
+    runs-on: ubuntu-latest
     steps:
       - uses: iterative/setup-cml@v1
       - uses: iterative/setup-dvc@v1
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
+          ref: ${{ github.event.pull_request.head.sha }}
       - name: Generate metrics report
         env:
           REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           cml ci
-          if [ $GITHUB_REF = refs/heads/main ]; then
+          if test ${{ github.ref_name }} = main; then
```

```suggestion
          if test $GITHUB_REF_NAME == main; then
```

Alternative spelling ™️